File: Python/Algorithms/1507.py
class Solution:
    months = [
        "Jan", "Feb", "Mar", "Apr",
        "May", "Jun", "Jul", "Aug",
        "Sep", "Oct", "Nov", "Dec"
    ]

    def reformatDate(self, date: str) -> str:
        day, month, year = date.split(' ')
        day, month = day[:-2], self.months.index(month) + 1
        return "-".join([
            year,
            f'0{month}' if month < 10 else f'{month}',
            day if len(day) == 2 else f'0{day}'
        ])

File: Javascript/Algorithms/1507.js
let months = new Map([
    "Jan", "Feb", "Mar", "Apr", 
    "May", "Jun", "Jul", "Aug", 
    "Sep", "Oct", "Nov", "Dec"
].map((x, idx) => [x, idx + 1]));

let reformatDate = (date) =>  {
    let [day, month, year] = date.split(' ');
        index = months.get(month);
    
    // Slice of th or rd.
    day = day.slice(0, -2);
    month = index < 10? `0${index}`: index;
    return [
        year, 
        month, 
        day.length == 2? day: `0${day}`
    ].join("-");
};
